pkg/registry/apis/datasource/register.go

```diff
@@ -191,6 +198,14 @@ func (b *DataSourceAPIBuilder) GetAPIGroupInfo(
 	return &apiGroupInfo, nil
 }
 
+func (b *DataSourceAPIBuilder) getPluginContext(ctx context.Context, uid string) (backend.PluginContext, error) {
```

Comment:
This feels like it adds even more complexity to get a plugin context than before. Understand you've done this to be able to separate the retrieval of plugin context and datasource instance settings with different interfaces. 

Not saying #80820 is perfect at all, but something along those lines would def help clear up all this madness with plugin context :)

Reply:
https://github.com/grafana/grafana/pull/80820 is fine -- but irrelevant to the problem here, and really internal detail anway so fine either way

Reply:
> separate the retrieval of plugin context and datasource instance settings with different interfaces.

exactly.  Mixing the two is required as the very last step.  Ideally as explicit as possible